website/src/docs/hotchocolate/defining-a-schema/object-types.md

```diff
@@ -0,0 +1,464 @@
+
+```csharp
+public class AuthorType : ObjectType<Author>
+{
+    protected override void Configure(IObjectTypeDescriptor<Author> descriptor)
+    {
+        descriptor.Name("BookAuthor");
+
+        descriptor
+            .Field(f => f.Name)
+            .Name("fullName");
+    }
+}
+```
+
+</ExampleTabs.Code>
+<ExampleTabs.Schema>
+
+Simply change the names in the schema.
+
+</ExampleTabs.Schema>
+</ExampleTabs>
+
+This would produce the following `BookAuthor` schema object type:
+
+```sdl
+type BookAuthor {
+  fullName: String
+}
+```
+
+# Explicit types
+
+Hot Chocolate will most of the type correctly infer the schema types of our fields. Sometimes we might have to be explicit about it though, for example when we are working with custom scalars.
+
+We can define explicit schema types like the following.
+
+<ExampleTabs>
+<ExampleTabs.Annotation>
+
+Per default the name of the class is the name of the object type in the schema and the names of the properties are the names of the fields of that object type.
+
+We can override these defaults using the `[GraphQLName]` attribute.
+
+```csharp
+public class Author
+{
+    [GraphQLType(typeof(StringType))]
+    public string Name { get; set; }
+}
+```
+
+</ExampleTabs.Annotation>
+<ExampleTabs.Code>
+
+In the Code-first approach we can use the `Type<T>` method on the `IObjectFieldDescriptor`.
+
+```csharp
+public class AuthorType : ObjectType<Author>
+{
+    protected override void Configure(IObjectTypeDescriptor<Author> descriptor)
+    {
+        descriptor
+            .Field(f => f.Name)
+            .Type<StringType>();
+    }
+}
+```
+
+</ExampleTabs.Code>
+<ExampleTabs.Schema>
+
+Simply change the field type in the schema.
+
+</ExampleTabs.Schema>
+</ExampleTabs>
+
+# Additional fields
+
+We can add additional (dynamic) fields to our schema types, without adding new properties to our backing class.
+
+<ExampleTabs>
+<ExampleTabs.Annotation>
+
+```csharp
+public class Author
+{
+    public string Name { get; set; }
+
+    public DateTime AdditionalField()
+    {
+        return DateTime.Now;
+    }
+}
+```
+
+</ExampleTabs.Annotation>
+<ExampleTabs.Code>
+
+In the Code-first approach we can use the `Type<T>` method on the `IObjectFieldDescriptor`.
+
+```csharp
+public class AuthorType : ObjectType<Author>
+{
+    protected override void Configure(IObjectTypeDescriptor<Author> descriptor)
+    {
+        descriptor
+            .Field("AdditionalField")
+            .Resolve(context =>
+            {
+                return DateTime.Now;
+            })
+    }
+}
+```
+
+</ExampleTabs.Code>
+<ExampleTabs.Schema>
+
+TODO
+
+</ExampleTabs.Schema>
+</ExampleTabs>
+
+What we have just created is a resolver. Hot Chocolate automatically creates resolvers for our properties, but we can also define them ourselves.
+
+Head over to the [resolver documentation](/docs/hotchocolate/fetching-data) to learn more.
```

Comment:
Do we need to touch on here (unless covered somewhere else):
* `ResolveWith`
* `Field<ClassWithResolvers>(x => x.GetSomeField(default!, default!, default!));`
* And maybe touch on the async overloads which some people struggle with too

Reply:
I plan on covering this in the resolver document.